### PR TITLE
Instrumentation with DD in react branch

### PR DESCRIPTION
**Pull request by timotheeg**
## Problem
We need to have deep instrumentation, tracing, and RUM prior to react launch

We don't want to setup DataDog RUM into angular, we can maintain sentry for that, but datadog RUM will be setup for React.

dd-trace is active, to work in containers, the EB environment **must** have the environment variable `DD_AGENT_HOST`.

## Solution
Activate datadog for everything we need:
* RUM on frontend
* APM on server
* Statsd

## References:
* [DD RUM intro and setup](https://docs.datadoghq.com/real_user_monitoring/browser/)
* [DD APM intro and setup for nodejs](https://docs.datadoghq.com/tracing/setup_overview/setup/nodejs/?tab=containers)

### CODE REVIEW

File: src/app/loaders/express/helmet.ts [hunk 1]
```diff
@@ -54,6 +54,7 @@ const helmetMiddlewares = () => {
       "'self'",
       'https://www.google-analytics.com/',
       'https://ssl.google-analytics.com/',
+      'https://*.browser-intake-datadoghq.com',
```
Comment (mantariksh) [inline src/app/loaders/express/helmet.ts:57]: can link to the docs where they list this domain

File: src/app/loaders/express/helmet.ts [hunk 2]
```diff
@@ -72,6 +73,9 @@ const helmetMiddlewares = () => {
       'https://www.gstatic.cn/',
       "'unsafe-inline'",
     ],
+    workerSrc: [
+      'blob:', // DataDog RUM session replay
```
Comment (mantariksh) [inline src/app/loaders/express/helmet.ts:77]: link to docs again

File: src/app/loaders/express/helmet.ts (hunk 1 above)
Comment (timotheeg) [reply src/app/loaders/express/helmet.ts]: https://github.com/opengovsg/FormSG/pull/3887/commits/ae7345811979a47b4be5f05b0eb1305f044362c7

File: src/app/loaders/express/helmet.ts (hunk 2 above)
Comment (timotheeg) [reply src/app/loaders/express/helmet.ts]: https://github.com/opengovsg/FormSG/pull/3887/commits/ae7345811979a47b4be5f05b0eb1305f044362c7

File: frontend/src/index.tsx [hunk 3]
```diff
@@ -14,6 +15,24 @@ if (process.env.NODE_ENV === 'test') {
...
+  applicationId: process.env.REACT_APP_DD_RUM_APP_ID || '',
+  clientToken: process.env.REACT_APP_DD_RUM_CLIENT_TOKEN || '',
+  env: process.env.REACT_APP_DD_RUM_ENV || '',
+  site: 'datadoghq.com',
+  service: 'formsg-react',
+
+  // Specify a version number to identify the deployed version of your application in Datadog
+  // version: '1.0.0',
+  sampleRate: 100,
+  replaySampleRate: 100,
```
Comment (timotheeg) [inline frontend/src/index.tsx:29]: Reduce this to 10, 10. 

Also check if we can inject a version in the init call

Comment (timotheeg) [reply frontend/src/index.tsx:29]: make the session sample be the reverse of react rollout percentage.

**Comment by karrui** [comment]
@timotheeg does this need to be updated due to conflicts? please request review from me when conflicts are fixed~

**Comment by timotheeg** [comment]
@karrui : Conflicts fixed, PR is ready for review

**Review by karrui — APPROVED** [review]
lgtm

**Pull request merged**